core/src/test/java/org/elasticsearch/index/query/FuzzyQueryBuilderTests.java

```diff
@@ -181,4 +187,16 @@ public void testParseFailsWithMultipleFields() throws IOException {
         e = expectThrows(ParsingException.class, () -> parseQuery(shortJson));
         assertEquals("[fuzzy] query doesn't support multiple fields, found [message1] and [message2]", e.getMessage());
     }
+
+    public void testDeprecationLoggingHeaders() throws IOException {
+        try (ThreadContext threadContext = new ThreadContext(Settings.EMPTY)) {
```

Comment:
I am on the fence. If other tests are executed concurrently, then adding the ThreadContext to the DeprecationLogger in the base class will mean that it potentially receives deprecation warnings from elsewhere. I was hoping to make this as local as possible so I'm almost sure that we test the following contructor call in isolation. So I'd rather add the context here, and even remove it from the Deprecation logger after this test is done. I'll push an update along that line.